composer/loggers/mosaicml_logger.py

```diff
@@ -96,10 +125,104 @@ def log_hyperparameters(self, hyperparameters: Dict[str, Any]):
     def log_metrics(self, metrics: Dict[str, Any], step: Optional[int] = None) -> None:
         self._log_metadata(metrics)
 
+    def log_analytics(self,) -> None:
+        if self.analytics_data is None:
+            return
+
+        # Fetch / cast metrics that we want to log from self.analytics_data
+        autoresume: bool = self.analytics_data['autoresume']
+        trainer_state: State = self.analytics_data['state']
+        save_interval: Union[str, int, Time, Callable[[State, Event], bool]] = self.analytics_data['save_interval']
+        loggers: List[LoggerDestination] = self.analytics_data['loggers']
+        load_path: Union[str, None] = self.analytics_data['load_path']
+        save_folder: Union[str, None] = self.analytics_data['save_folder']
+
+        metrics: Dict[str, Any] = {'composer/autoresume': autoresume, 'composer/precision': trainer_state.precision}
+
+        train_dataloader = trainer_state.train_dataloader
+        if train_dataloader is not None and isinstance(train_dataloader, torch.utils.data.DataLoader):
+            metrics['composer/train_loader_workers'] = train_dataloader.num_workers
+
+        metrics['composer/eval_loaders'] = []
+        for evaluator in trainer_state.evaluators:
+            dataloader = evaluator.dataloader.dataloader
+            if isinstance(dataloader, torch.utils.data.DataLoader):
+                metrics['composer/eval_loaders'].append(
+                    json.dumps({
+                        'label': evaluator.label,
+                        'num_workers': dataloader.num_workers,
+                    }),
+                )
+
+        get_optimizer_args = lambda optimizer: {
```

Comment:
this is a bit dangerous (also please define as a private function in this file rather than a lambda). I suggest you just get `optimizer.defaults`